### PR TITLE
PIM-9540: do not strip HTML tags on textara content before index them in ES

### PR DESCRIPTION
**Pull request by olivierpontier**
**Description (for Contributor and Core Developer)**

With this pull request we do not strip HTML tags anymore before indexing a textarea content in Elasticsearch.
Based on the [Java Regular expression docimentation](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html), we fixed the newline_pattern char filter which is used in the textarea analyzer. The html_strip char filter replaces HTML block tags (`<p>`, `<div>`, ...) by a newline character. Now, the newline_pattern char filter takes account correctly of those additional newline characters.
We improved our integration tests on textarea to validate the searching on textarea attributes with HTML content.

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

**Comment by olivierpontier** [comment]
Index the whole HTML content has impacts on the use of the folloing search filters: StartWIth, EndsWith, EqualsTo and NotEqualsTo.

File: tests/back/Pim/Enrichment/Integration/PQB/Filter/Attribute/TextArea/TextAreaFilterIntegration.php [hunk 1]
```diff
@@ -168,7 +168,7 @@ public function testOperatorEquals()
                 'my <bold>cat</bold> is the most <i>beautiful</i><br/>',
             ],
         ]);
-        $this->assert($result, []);
+        $this->assert($result, ['best_cat']);
```
Comment (jmleroux) [inline tests/back/Pim/Enrichment/Integration/PQB/Filter/Attribute/TextArea/TextAreaFilterIntegration.php:171]:
```suggestion
        $this->assert(['best_cat'], $result);
```

Comment (jmleroux) [reply tests/back/Pim/Enrichment/Integration/PQB/Filter/Attribute/TextArea/TextAreaFilterIntegration.php:171]: Same for all tests, expected should be first.

**Review by jmleroux — APPROVED** [review]
GTM wth one comment, take it or not.

File: tests/back/Pim/Enrichment/Integration/PQB/Filter/Attribute/TextArea/TextAreaFilterIntegration.php (hunk 1 above)
Comment (olivierpontier) [reply tests/back/Pim/Enrichment/Integration/PQB/Filter/Attribute/TextArea/TextAreaFilterIntegration.php:171]: It is done here: https://github.com/akeneo/pim-community-dev/pull/13181/files/9881547eaa35126c381b9a009eb7468d214feed1#diff-1545b575ff9ef9aec731772144704b4eaad9951b60f38b094ac3b3ec47c4b477R164

**Review by BitOne — APPROVED** [review]

**Pull request merged**